engine/engine/content/builtins/scripts/debugger.lua

```diff
@@ -63,28 +63,30 @@ local function remove_breakpoint(line, file)
   end
 end
 
+local file_cache = {}
 -- This hook fires only on Lua functions calls and checks if this function has lines
 -- we have breakpoints on
-local function lightweight_hook()
-  local caller = debug.getinfo(2,"S")
-  local file = mobdebug.get_filename(caller.source)
+local function lightweight_hook(source, lastlinedefined, co)
+  local file = file_cache[source]
+  if not file then 
+    file = mobdebug.get_filename(source)
+    file_cache[source] = file
+  end
   mobdebug.get_server_data(file)
   local brk = breakpoints[file]
-  local co = coroutine.running()
   local in_co = false
   if co then
     -- If it's the first call after the [C] call - save function metricks
-    if caller.lastlinedefined ~= -1 and coroutines[co] == false then
+    if lastlinedefined ~= -1 and coroutines[co] == false then
         -- Coroutine provide this info only on the first coroutine fn call
         -- we should save it for later usage
         coroutines[co] = {
           ["lines"] = debug.getinfo(2,"L"),
-          ["file"] = file,
-          ["caller"] = caller
+          ["file"] = file
         }
     end
     -- Make sure [C] function call was before the first coroutine call
-    if not coroutines[co] and caller.lastlinedefined == -1 then
+    if not coroutines[co] and lastlinedefined == -1 then
```

Comment:
But, how does this work?
Who owns/sets the `lastlinedefined` now?

Reply:
now we get it on cpp side and use as function argument :
`local function lightweight_hook(source, lastlinedefined, co)`